Sources/GRPC/TLSConfiguration.swift

```diff
@@ -0,0 +1,195 @@
+  /// specification.
+  public struct TLS {
+    public private(set) var configuration: TLSConfiguration
+
+    /// Value to use for TLS SNI extension; this must not be an address.
+    public var hostnameOverride: String?
+
+    /// The certificates to offer during negotiation. If not present, no certificates will be offered.
+    public var certificateChain: [NIOSSLCertificateSource] {
+      get {
+        return self.configuration.certificateChain
+      }
+      set {
+        self.configuration.certificateChain = newValue
+      }
+    }
+
+    /// The private key associated with the leaf certificate.
+    public var privateKey: NIOSSLPrivateKeySource? {
+      get {
+        return self.configuration.privateKey
+      }
+      set {
+        self.configuration.privateKey = newValue
+      }
+    }
+
+    /// The trust roots to use to validate certificates. This only needs to be provided if you
+    /// intend to validate certificates.
+    public var trustRoots: NIOSSLTrustRoots? {
+      get {
+        return self.configuration.trustRoots
+      }
+      set {
+        self.configuration.trustRoots = newValue
+      }
+    }
+
+    /// Whether to verify remote certificates.
+    public var certificateVerification: CertificateVerification {
+      get {
+        return self.configuration.certificateVerification
+      }
+      set {
+        self.configuration.certificateVerification = newValue
+      }
+    }
+
+    /// TLS Configuration with suitable defaults for clients.
+    ///
+    /// This is a wrapper around `NIOSSL.TLSConfiguration` to restrict input to values which comply
+    /// with the gRPC protocol.
+    ///
+    /// - Parameter certificateChain: The certificate to offer during negotiation, defaults to an
+    ///     empty array.
+    /// - Parameter privateKey: The private key associated with the leaf certificate. This defaults
+    ///     to `nil`.
+    /// - Parameter trustRoots: The trust roots to validate certificates, this defaults to using a
+    ///     root provided by the platform.
+    /// - Parameter certificateVerification: Whether to verify the remote certificate. Defaults to
+    ///     `.fullVerification`.
+    /// - Parameter hostnameOverride: Value to use for TLS SNI extension; this must not be an IP
+    ///     address, defaults to `nil`.
+    public init(
+      certificateChain: [NIOSSLCertificateSource] = [],
+      privateKey: NIOSSLPrivateKeySource? = nil,
+      trustRoots: NIOSSLTrustRoots = .default,
+      certificateVerification: CertificateVerification = .fullVerification,
+      hostnameOverride: String? = nil
+    ) {
+      self.configuration = .forClient(
+        minimumTLSVersion: .tlsv12,
+        certificateVerification: certificateVerification,
+        trustRoots: trustRoots,
+        certificateChain: certificateChain,
+        privateKey: privateKey,
+        applicationProtocols: GRPCApplicationProtocolIdentifier.allCases.map { $0.rawValue }
```

Comment:
Sorry, which identifiers does this include again?

Reply:
Off the top of my head: "h2" and "grpc-ext"